test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -67,12 +67,15 @@ timestamp_manager::decimal_to_hex(const uint64_t ts)
 }
 
 int
-timestamp_manager::validate_hex_value(const std::string &ts_string)
+timestamp_manager::validate_hex_value(const std::string &ts_string, const std::string ts)
 {
     /* Check that the timestamp string has valid hexadecimal characters. */
     for (auto &ch : ts_string)
         if (!std::isxdigit(ch))
-            WT_SIM_RET_MSG(EINVAL, "Illegal commit timestamp: invalid hex value.");
+            WT_SIM_RET_MSG(EINVAL, "Illegal " + ts + " passed: invalid hex value.");
+
+    if (ts_string == "0")
```

Comment:
Should we check for negative values too?

Reply:
The previous check will ensure that for negative values we output an error.